firmware/gameport-adapter/ThrustmasterAttackThrottle.h

```diff
@@ -0,0 +1,370 @@
+class AnalogAxisState {
+public:
+  AnalogAxisState()
+  : m_min()
+  , m_max() {
+  }
+
+  void init(uint8_t startPos) {
+    m_min = startPos;
+    m_max = startPos;
+  }
+
+  void init(uint8_t min, uint8_t max) {
+    m_min = min;
+    m_max = max;
+  }
+
+  int update(uint8_t analogValue) {
+    if (analogValue < m_min) {
+      m_min = analogValue;
+    } else if (analogValue > m_max) {
+      m_max = analogValue;
+    }
+
+    return map(analogValue, m_min, m_max, 0, 1023);
+  }
+
+private:
+  uint8_t m_min;
+  uint8_t m_max;
+};
+} // namespace
+
+/// The Thrustmaster Attack Throttle uses the all-digital DirectConnect protocol
+/// It has an input port that any analog joystick can be plugged into. But it was
+/// usually used with a flight stick and rudder pedals. Thrustmaster sold a kit
+/// that included the throttle with the Top Gun analog joystick and the Elite Rudder Pedals.
+///
+/// DirectConnect supported multiple digital joysticks on the same port
+/// The Linux driver supports up to 2. This code doesn't currently support
+/// more than 1 as I only have one digital device to test with.
+///
+/// This assumes the analog joystick plugged in matches the thrustmaster
+/// configuration for the Top-Gun joystick and Elite Rudder Pedals.
+/// It is fine to use something different such as just the joystick
+/// but there is Thrustmaster-specific handling of the joystick hat.
+class ThrustmasterAttackThrottle : public Joystick {
+public:
+  /// Resets the joystick and tries to detect the model.
+  bool init() override {
+    uint8_t packet[packetSize];
+    bool packetOk = false;
+
+    // The trigger pin starts out high, this causes two edges
+    // on the first readPacket which throws off the timing.
+    // Force to low first, allow it to stablize, then start
+    // the attempt.
+    m_trigger.setLow();
+    delay(100);
+
+    // 10 times is excessive, it should always work 100%
+    // but with how timing sensitive the code is, I wanted to
+    // give it some tries.
+    for (int i = 0; i < 10 && !packetOk; ++i) {
+      packetOk = readPacket(packet);
+      if (!packetOk) {
+        delay(200);
+      }
+    }
+
+    // found an ok packet, initialize the axis range state
+    if (packetOk) {
+      for (size_t axis = 0; axis < axisIndex_size; ++axis) {
+        uint8_t analogValue = packet[axisIndex[axis]];
+
+        // Use the throttle ranges for the throttle,
+        // otherwise treat the initial value as the center.
+        if (axis == throttleAxis) {
+          m_axisStates[axis].init(throttleMin, throttleMax);
+        } else {
+          m_axisStates[axis].init(analogValue);
+        }
+      }
+    }
+    return packetOk;
+  }
+
+  bool update() override {
+    uint8_t packet[packetSize];
+
+    bool packetOk = readPacket(packet);
+    if (packetOk) {
+      updateState(packet);
+    } else {
+      // wait a bit between bad packets in case it is still trying to output data.
+      delay(200);
+    }
+    delay(53);
```

Comment:
The Thrustmaster drivers use a 55ms delay between triggers. The data itself is about 1.5 ms in duration and is sent after a 0.5ms delay. So that leaves 53ms sitting there. From my captures most of the time the bus is idle. I can try increasing the pace but I don't have any way to know that will always work, vs just happens to work with my joystick. I suspect the large delay is to support multiple digital devices plugged into the same port but i'm sure some portion is needed. I just don't know what portion is required vs what the drivers do.